contracts/signature/Eip712Signature.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.0 <0.9.0;
+
+/// @title EIP712Signature - A contract that extracts and inspects EIP-712 signatures appended to calldata.
+
+abstract contract EIP712Signature {
+    uint256 private nonce;
+
+    /**
+     * @dev Returns the current nonce value.
+     */
+    function eip712Nonce() public view returns (uint256) {
+        return nonce;
+    }
+
+    /**
+     * @dev Increments nonce.
+     */
+    function eip712BumpNonce() internal {
+        nonce = nonce + 1;
+    }
+
+    /**
+     * @dev When signature present in calldata, returns the address of the signer.
+     */
+    function eip712SignedBy() internal returns (address signer) {
+        if (msg.data.length >= 4 + 32 + 32 + 1) {
```

Comment:
Is there any restriction that the call should be done with `msg.data.length>=4`? I mean, the structure forbids calls of fallback functions?

Reply:
Here we are checking whether there is at least 65 bytes at end:
* If yes we use those bytes to produce a signature recovery
* If not we return zero address

Answers the question, or have I missunderstood?

Reply:
I've in the meantime reworked the Mod to include EIP-1271. I think I cleared up the "dataTrimmed" situation. Does your question still persist?

Reply:
@Barichek, in this case, the user will always be signing a message to a specific function selector on the module (probably `execTransactionFromModul()`), no need for fallback support.

Reply:
I see, thank you.